brave/src/main/java/brave/propagation/CorrelationField.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.propagation;
+
+import brave.internal.Nullable;
+
+/**
+ * A field which are applied in a correlation context such as MDC via {@link
+ * CorrelationFieldScopeDecorator}.
+ *
+ * <p>Field updates only apply during {@linkplain CorrelationFieldScopeDecorator scope
+ * decoration}. This means values set do not flush immediately to the underlying correlation context
+ * by default. Rather, they are scheduled for the next scope operation. This is a way to control
+ * overhead. Use the type {@link Flushable} to allow immediate updates.
+ *
+ * <p>{@link #equals(Object)} and {@link #hashCode()} should be overridden to implement lower-case
+ * {@link #name()} comparison.
+ *
+ * @see CorrelationFieldScopeDecorator
+ * @since 5.11
+ */
+public interface CorrelationField {
+  /** The name of this field in the correlation context. */
+  String name();
+
+  /**
+   * Returns the most recent value of the field named {@link #name()} in the context or null if
+   * unavailable.
+   */
+  @Nullable String getValue(TraceContext context);
+
+  /**
+   * Marks a field such that updates immediately flush to the correlation context, when configured
+   * and their corresponding trace context is current. This has a significant performance impact as
+   * it will invalidate the {@link CurrentTraceContext#maybeScope(TraceContext)} operator.
```

Comment:
I rewrote as this:

```
     * Call this to immediately flush a value update to the correlation context as opposed waiting
     * for the next scope decoration. This has a significant performance impact as it requires even
     * {@link CurrentTraceContext#maybeScope(TraceContext)} to always track values.
```

maybeScope returns no-op when applying the context results in the same data. maybeScope when not interfered with prevents another layer (ex several arrays). If we don't know if a field will be flushed or not, we need to allocate all of that even if no change ever takes place. This could happen thousands of times.